driver/bpf/filler_helpers.h

```diff
@@ -73,55 +70,159 @@ static __always_inline struct file *bpf_fget(int fd)
 	return fil;
 }
 
-// Kernel 5.10 introduced a new bpf_helper called `bpf_d_path` to extract a file path starting from a file descriptor.
-// Libscap loads our bpf programs as `BPF_PROG_TYPE_RAW_TRACEPOINT` programs. This type of program doesn't seem able to call this new helper because it is out of its scope. For more details see here https://github.com/torvalds/linux/blob/58e1100fdc5990b0cc0d4beaf2562a92e621ac7d/kernel/trace/bpf_trace.c#L1574
-static __always_inline char *bpf_get_path(struct filler_data *data, int fd)
-{
-	struct file *f = bpf_fget(fd);
-	const unsigned char** pointers_buf = (const unsigned char**)data->tmp_scratch;
-	char *filepath = (char *)&data->tmp_scratch[(MAX_PATH_COMPONENTS* sizeof(const unsigned char*)) & SCRATCH_SIZE_HALF];
+/* In this kernel version the instruction limit was bumped from 131072 to 1000000.
+ * For this reason we use different values of `MAX_NUM_COMPONENTS`
+ * according to the kernel version.
+ */
+#if(LINUX_VERSION_CODE >= KERNEL_VERSION(5, 2, 0))
+#define MAX_NUM_COMPONENTS 48
+#else
+#define MAX_NUM_COMPONENTS 24
+#endif
 
-	struct dentry *de_p = _READ(f->f_path.dentry); 
-	if(!de_p)
+/* We must always leave at least 4096 bytes free in our tmp scratch space
+ * to please the verifier since we set the max component len to 4096 bytes.
+ * We start writing our exepath from half of the tmp scratch space. The
+ * whole space is 256 KB, we start at 128 KB.
+ *
+ *       128 KB           128 KB (Free space to please the verifier)
+ * |----------------|----------------|
+ *                  ^
+ *                  |
+ *        We start here and write backward
+ *        as we find components of the path
+ *
+ * As a bitmask we use `SAFE_TMP_SCRATCH_ACCESS` (128*1024 - 1).
+ * This helps the verifier to understand that our offset never overcomes
+ * 128 KB.
+ */
+#define MAX_COMPONENT_LEN 4096
+#define MAX_TMP_SCRATCH_LEN (SCRATCH_SIZE >> 1)
+#define SAFE_TMP_SCRATCH_ACCESS(x) (x) & (MAX_TMP_SCRATCH_LEN - 1)
+
+/* Please note: Kernel 5.10 introduced a new bpf_helper called `bpf_d_path`
+ * to extract a file path starting from a struct* file but it can be used only
+ * with specific hooks:
+ *
+ * https://github.com/torvalds/linux/blob/e0dccc3b76fb35bb257b4118367a883073d7390e/kernel/trace/bpf_trace.c#L915-L929.
```

Comment:
In addition, @Andreagit97 could you reference @loresuso comment and alternative proposition outlined in https://github.com/falcosecurity/libs/issues/1111 in both the bpf and modern bpf drivers in order to give some credit to Lorenzo's research :) ? 

One day we should still explore @loresuso suggestions. Meanwhile knowing about the limitations listed below the fact that we perform the lookups in the execve* calls likely is still most robust at the end ...

Reply:
done :)